From 5c3b9440cabb1c61d2afd2adf9444a190f2f835e Mon Sep 17 00:00:00 2001
From: Ross Lagerwall <ross.lagerwall@citrix.com>
Date: Mon, 3 Apr 2017 09:02:16 +0100
Subject: [PATCH] CA-248714: Don't fail to start if no IPv4 address is
 configured

Partially backport 3de3d698d942 ("util/qemu-sockets: improve ai_flag
hints for ipv6 hosts") which removes AI_ADDRCONFIG from the
getaddrinfo() options.  AI_ADDRCONFIG causes getaddrinfo() to fail
unless the host has an IPv4 address. This isn't useful for listening on
127.0.0.1 for VNC connections.

From the logs, it appears that this failure occurs when starting QEMU
while the host is renewing the DHCP lease.
---
 qemu-sockets.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/qemu-sockets.c b/qemu-sockets.c
index 612d922..f8094ea 100644
--- a/qemu-sockets.c
+++ b/qemu-sockets.c
@@ -102,7 +102,7 @@ int inet_listen(const char *str, char *ostr, int olen,
     int slisten,rc,pos,to,try_next;
 
     memset(&ai,0, sizeof(ai));
-    ai.ai_flags = AI_PASSIVE | AI_ADDRCONFIG;
+    ai.ai_flags = AI_PASSIVE;
     ai.ai_family = PF_UNSPEC;
     ai.ai_socktype = socktype;
 
-- 
2.7.4

